contracts/WildcardSteward_v2.sol

```diff
@@ -467,16 +481,59 @@ contract WildcardSteward_v2 is Initializable {
         address tokenPatron = currentPatron[tokenId];
 
         if (state[tokenId] == StewardState.Owned) {
-            uint256 totalToPayBack = price[tokenId];
-            // NOTE: pay back the deposit only if it is the only token the patron owns.
-            if (
-                totalPatronOwnedTokenCost[tokenPatron] ==
-                price[tokenId].mul(patronageNumerator[tokenId])
-            ) {
-                totalToPayBack = totalToPayBack.add(deposit[tokenPatron]);
-                deposit[tokenPatron] = 0;
-            }
+            _buy(tokenId, tokenPatron, wildcardsPercentage);
+        } else if (state[tokenId] == StewardState.Foreclosed) {
+            _buyForeclosure(tokenId);
+        }
+
+        wildcardsPercentages[tokenId] = wildcardsPercentage; //Set the percentage wildcards will recieve on next sale
+        deposit[msg.sender] = deposit[msg.sender].add(remainingValueForDeposit);
+        transferAssetTokenTo(
+            tokenId,
+            currentOwner,
+            tokenPatron,
+            msg.sender,
+            _newPrice
+        );
+        emit Buy(tokenId, msg.sender, _newPrice);
+    }
+
+    // NB to check that the msg.value is greater tgat
+    function _buyForeclosure(uint256 tokenId) internal {
+        uint256 totalAmount = price[tokenId];
+        uint256 artistAmount = totalAmount.mul(100).div(10000); //1% hard coded at the moment
+        uint256 wildcardsAmount = totalAmount.sub(artistAmount);
+
+        _payArtistAndWildcards(tokenId, artistAmount, wildcardsAmount);
 
+        state[tokenId] = StewardState.Owned;
+        timeLastCollected[tokenId] = now;
+        timeLastCollectedPatron[msg.sender] = now;
+    }
+
+    function _buy(
+        uint256 tokenId,
+        address tokenPatron,
+        uint256 wildcardsPercentage
+    ) internal {
+        uint256 totalAmount = price[tokenId];
+        if (wildcardsPercentages[tokenId] == 0) {
```

Comment:
For all the existing tokens, thought that it may be easier to just check if this value hasn't been set and make it a 5% default. Otherwise we could create another function to add percentages to all the existing tokens, in the long run maybe better as less gas since this doesn't have to be checked on every buy, but obviously a lot more cumbersome.